### ISSUE TITLE
[Step Functions] False positive when using JSONata in Map field

### ISSUE THREAD

**Issue by AlexChesters** [aws-cloudformation/cfn-lint#4096]
### CloudFormation Lint Version

cfn-lint 1.33.1

### What operating system are you using?

Mac

### Describe the bug

False positive violation of `E3601` when using Map in a step function with JSONata

### Expected behavior

A valid template should not produce errors

### Reproduction template

```yaml
Resources:
  Role:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              Service:
                - states.amazonaws.com
            Action:
              - sts:AssumeRole
  StateMachine:
    Type: AWS::StepFunctions::StateMachine
    Properties:
      StateMachineName: my-step-function
      RoleArn: !GetAtt Role.Arn
      Definition:
        QueryLanguage: JSONata
        StartAt: Assign variables
        States:
          Assign variables:
            Type: Pass
            Assign:
              account_id: '{% $states.input.account_id %}'
            Next: ListPermissionSetsProvisionedToAccount
          ListPermissionSetsProvisionedToAccount:
            Type: Task
            Arguments:
              AccountId: "{% $account_id %}"
              InstanceArn: "arn:aws:sso:::instance/ssoins-aaaaaaaaaaaa"
            Resource: "arn:aws:states:::aws-sdk:ssoadmin:listPermissionSetsProvisionedToAccount"
            Next: "Delete permission sets"
            Credentials:
              RoleArn: "arn:aws:iam::111111111111:role/aws-accounts-lifecycle-account-resetter"
            Output:
              permission_sets: "{% $states.result.PermissionSets %}"
          Delete permission sets:
            Type: Map
            ItemProcessor:
              ProcessorConfig:
                Mode: INLINE
              StartAt: Pass
              States:
                Pass:
                  Type: Pass
                  End: true
            End: true
            Items: "{% $states.input.permission_sets %}"
            MaxConcurrency: 1
```
